src/quantum/azext_quantum/operations/workspace.py

```diff
@@ -0,0 +1,187 @@
+                return value
+            value = cmd.cli_ctx.config.get('quantum', key, None)
+            if value is not None:
+                return value
+            value = cmd.cli_ctx.config.get(cmd.cli_ctx.config.defaults_section_name, key, None)
+            return value
+
+        self.subscription = get_subscription_id(cmd.cli_ctx)
+        self.resource_group = select_value('group', resource_group_name)
+        self.name = select_value('workspace', workspace_name)
+        self.location = select_value('location', location)
+
+    def clear(self):
+        self.subscription = ''
+        self.resource_group = ''
+        self.name = ''
+        self.location = ''
+
+    def save(self, cmd):
+        from azure.cli.core.util import ConfiguredDefaultSetter
+
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, False):
+            cmd.cli_ctx.config.set_value('quantum', 'group', self.resource_group)
+            cmd.cli_ctx.config.set_value('quantum', 'workspace', self.name)
+            cmd.cli_ctx.config.set_value('quantum', 'location', self.location)
+
+
+def _get_storage_account_path(workspaceInfo, storage_account_name):
+    if (storage_account_name[0] == "/"):
+        path = storage_account_name
+    else:
+        path = f"/subscriptions/{workspaceInfo.subscription}/resourceGroups/{workspaceInfo.resource_group}/providers/Microsoft.Storage/storageAccounts/{storage_account_name}"
+    return path
+
+
+def _get_basic_quantum_workspace(location, info, storage_account):
+    qw = QuantumWorkspace(location=location)
+    # Use a default provider 
+    # Replace this with user specified providers as part of task 16184.
+    prov = Provider()
+    prov.provider_id = "Microsoft"
+    prov.provider_sku = "Basic"
+    qw.providers = [prov]
+    # Allow the system to assign the workspace identity
+    qw.identity = QuantumWorkspaceIdentity()
+    qw.identity.type = "SystemAssigned"
+    qw.storage_account = _get_storage_account_path(info, storage_account)
+    return qw
+
+
+def _create_role_assignment(cmd, quantum_workspace):
+    from azure.cli.command_modules.role.custom import create_role_assignment
+    retry_attempts = 0
+
+    while (retry_attempts < MAX_RETRIES_ROLE_ASSIGNMENT):
+        try:
+            create_role_assignment(cmd, role="Contributor", scope=quantum_workspace.storage_account, assignee=quantum_workspace.identity.principal_id)
+            break
+        except (CloudError, CLIError) as e:
+            error = str(e.args).lower()
+            if (("does not exist" in error) or ("cannot find" in error)):
+                print('.', end='', flush=True)
+                time.sleep(POLLING_TIME_DURATION)
+                retry_attempts += 1
+                continue
+            raise e
+        except Exception as x:
+            raise CLIError(f"Role assignment encountered exception ({type(x).__name__}): {x}")
+
+    if (retry_attempts > 0):
+        print("") # To end the line of the waiting indicators.
```

Comment:
What's the effect?

Reply:
Of what part?

Reply:
`print("")` does not have effect. What's the purpose of this line?

Reply:
It prints a new line. By default if `end` is not specified, then it will always append a new line, even to an empty string.

Reply:
You may use `print()` instead.

Reply:
Sorry. I muddle them. `printf()` in C language does not print a new line while `print()` in Python does.